File: MySecondApp/app/src/main/java/com/hfad/mysecondapp/MainActivity.java
package com.hfad.mysecondapp;

import android.os.Bundle;
import android.view.View;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private final BeerExpert beerExpert = new BeerExpert();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
    }

    public void onClickFindBeer(View view) {
        Spinner beer_type_spinner = findViewById(R.id.beer_spinner);
        String beer_type_str = String.valueOf(beer_type_spinner.getSelectedItem());
        TextView beers = findViewById(R.id.beers);
        List<String> corresponding_beers = beerExpert.getBeers(beer_type_str);
        StringBuilder beers_str = new StringBuilder();
        for (String beer : corresponding_beers) {
            beers_str.append(beer).append('\n');
        }
        beers.setText(beers_str);
    }
}

File: MyApplication/KOTLIN/01_firstChapter/app/src/main/java/com/headfirstandroiddev/app1/MainActivity.kt
package com.headfirstandroiddev.app1

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        supportActionBar?.title = "Find Drinks!"
        setContentView(R.layout.activity_main)
        val button1 = findViewById<Button>(R.id.button1_mainActivity)
        val spinner1 = findViewById<Spinner>(R.id.spinner1_mainActivity)
        button1.setOnClickListener { _ ->
            val selectedItemString= spinner1.selectedItem.toString()
            val textView1 = findViewById<TextView>(R.id.textView1_mainActivity)
            textView1.text = getDrinks(selectedItemString)
        }

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun getDrinks(item: String): String {
        return when (item) {
            ("Beer") -> "Guinness\nHeineken\nBudweiser"
            ("Wine") -> "Chardonnay\nSauvignon Blanc\nPinot Noir"
            ("Vodka") -> "Absolut\nChivas Regal\nSmirnoff"
            else -> "No drink selected!"
        }
    }
}